nemo_skills/code_execution/sandbox.py

```diff
@@ -357,7 +357,7 @@ def update_fn(map_to_future, line_dict, answer_format="natural_language"):
                 line_dict["is_correct"] = map_to_future[
                     (line_dict["predicted_answer"], line_dict["expected_answer"])
                 ].result()
-            elif answer_format == "lean4-proof" or answer_format == "lean4-statement":
+            elif answer_format == "lean4-proof" or answer_format == "lean4-statement" or anwer_format == "lean4-statement-with-header":
```

Comment:
This function is getting way too big and hard to understand. If you have some ideas on how to make it simpler, please give it a try. E.g. we can separate lean and math checks or try to modularize parts of this or just simplify the code by rewriting it in a different way

Reply:
Since all the lean4 answer format are in form lean4-*, maybe I could replace with "lean4" in answer_format?

Reply:
if that makes it simpler, let's do it